engine/core/src/main/java/org/datacleaner/metamodel/datahub/DataHubConnection.java

```diff
@@ -136,4 +143,11 @@ private URIBuilder appendToPath(final URIBuilder uriBuilder, final String pathSe
 
         return uriBuilder.setPath(pathSegment);
     }
+
+    private class DataHubRequestRetryHandler extends DefaultHttpRequestRetryHandler {
+        DataHubRequestRetryHandler() {
+            super(3, false,
+                    Arrays.asList(InterruptedIOException.class, UnknownHostException.class, SSLException.class));
```

Comment:
The reason I extend the DefaultHttpRequestRetryHandler using the constructor is that the standard constructors of that class adds ConnectException to the list of "nonRetriable" classes, but in this case we do want to retry in case of a ConnectException, because that's just what the timeout triggers.